builder/tests/test_actions.py

```diff
@@ -90,60 +90,175 @@ def test_duplicate_search(version_from_scratch):
+
+    # Assert...
+    # that the codelist has 2 searches
+    assert draft.searches.count() == 2
+    # that the codelist has 3 codes
+    assert draft.code_objs.count() == 3
+
+    # Act: delete the search for "Enthesopathy of elbow region"
+    actions.delete_search(search=s3)
+
+    # Assert...
+    # that the codelist has only 1 search
+    assert draft.searches.count() == 1
+    # that the codelist now has 2 codes; although codes 73583000 and 202855006 are
+    # in the search for "Enthesopathy of elbow region", they are not deleted because
+    # the are also in the search for "epicondylitis"
+    assert draft.code_objs.count() == 2
+
+    # Act: delete the remainng search for "epicondylitis"
+    actions.delete_search(search=s1)
+    assert draft.searches.count() == 0
+    assert draft.code_objs.count() == 0
+
+
+def test_delete_search_codelist_with_codes(version_with_no_searches):
+    draft = version_with_no_searches
+    codelist_codes = draft.codeset.all_codes()
+    included_codes = draft.codeset.codes()
+    assert len(codelist_codes) == 8
+    assert len(included_codes) == 6
+
+    # This codelist has all the codes for disorder of elbow excluding arthritis
+    # 8 codes, 6 included
+    """
+                        + 128133004 Disorder of elbow
+                                        |
+         --------------- --------------- ----------------------- -------------------
+        |                               |                       |                   |
+    + 429554009                     + 35185008              + 239964003           + 156659008
+    Arthropathy of elbow          Enthesopathy of       Soft tissue lesion     (Epicondylitis &/or tennis elbow)
+        |                           elbow region          of elbow region       or (golfers' elbow) [inactive]
+        |                               |
+    - 439656005                     + 73583000
+    Arthritis of elbow            Epicondylitis
+        |                               |
+        -----------202855006-------------
+              - Lateral epicondylitis
+    """
+    # Epicondylitis codes are already included on the codelist; one is included, one is not
+    epicondylitis_included_code = "73583000"
+    epicondylitis_excluded_code = "202855006"
+    epicondylitis_codes = {epicondylitis_included_code, epicondylitis_excluded_code}
+    # Tennis toe code is not on the codelist
+    tennis_toe_code = "238484001"
+
+    # Arthritis of elbow is on the codelist, but not included
+    arthritis_of_elbow_code = "439656005"
+
+    assert epicondylitis_included_code in included_codes
+    assert epicondylitis_excluded_code not in included_codes
+    for code in epicondylitis_codes:
+        assert code in codelist_codes
+    assert tennis_toe_code not in codelist_codes
+    assert tennis_toe_code not in included_codes
+    assert arthritis_of_elbow_code in codelist_codes
+    assert arthritis_of_elbow_code not in included_codes
```

Comment:
I'm sorry you've had to do this!  Do you think it'd be worth spending time to produce some fixtures that are more legible?